model_api/python/openvino/model_api/models/anomaly.py

```diff
@@ -0,0 +1,152 @@
+
+        Returns:
+            AnomalyResult: Results
+        """
+        anomaly_map: np.ndarray | None = None
+        pred_label: str | None = None
+        pred_mask: np.ndarray | None = None
+        pred_boxes: np.ndarray | None = None
+        box_labels: np.ndarray | None = None
+
+        predictions = outputs[list(self.outputs)[0]]
+
+        if len(predictions.shape) == 1:
+            pred_score = predictions
+        else:
+            anomaly_map = predictions.squeeze()
+            pred_score = anomaly_map.reshape(-1).max()
+
+        pred_label = "Anomalous" if pred_score > self.image_threshold else "Normal"
+
+        if self.task in ("segmentation", "detection"):
+            assert anomaly_map is not None  # for mypy
+            pred_mask = (anomaly_map >= self.pixel_threshold).astype(np.uint8)
+            anomaly_map = self._normalize(anomaly_map, self.pixel_threshold)
+
+        # normalize
+        pred_score = self._normalize(pred_score, self.image_threshold)
+
+        # resize outputs
+        if anomaly_map is not None:
+            anomaly_map = cv2.resize(
+                anomaly_map, (meta["original_shape"][1], meta["original_shape"][0])
+            )
+            pred_mask = cv2.resize(
+                pred_mask, (meta["original_shape"][1], meta["original_shape"][0])
+            )
+
+        if self.task == "detection":
+            pred_boxes = self._get_boxes(pred_mask)
+            box_labels = np.ones(pred_boxes.shape[0])
+
+        return AnomalyResult(
+            anomaly_map=anomaly_map,
+            box_labels=box_labels,
+            pred_boxes=pred_boxes,
+            pred_label=pred_label,
+            pred_mask=pred_mask,
+            pred_score=pred_score.item(),
+        )
+
+    @classmethod
+    def parameters(cls) -> Dict:
+        parameters = super().parameters()
+        parameters.update(
+            {
+                "image_shape": ListValue(
+                    description="Image shape",
+                ),
+                "image_threshold": NumericalValue(
+                    description="Image threshold", min=0.0, default_value=0.5
+                ),
+                "pixel_threshold": NumericalValue(
+                    description="Pixel threshold", min=0.0, default_value=0.5
+                ),
+                "max": NumericalValue(
+                    description="max value for normalization",
+                ),
+                "min": NumericalValue(
+                    description="min value for normalization",
+                ),
+                "task": StringValue(
+                    description="Task type", default_value="segmentation"
+                ),
+            }
+        )
+        return parameters
+
+    def _normalize(self, tensor: np.ndarray, threshold: float) -> np.ndarray:
+        """Currently supports only min-max normalization."""
+        normalized = ((tensor - threshold) / (self.max - self.min)) + 0.5
```

Comment:
This is the only usage of `self.max` and `self.min`. Consider merging them into one parameter since you only need `self.max - self.min`